### PR TITLE
fix(pkg): properly support "!" for breaking changes.

### PR DESCRIPTION
**Pull request by FedeDP**
Moreover, "feat" is also a valid feature prefix name.

### CODE REVIEW

File: pkg/releasenotes/print.go [hunk 1]
```diff
@@ -80,24 +80,18 @@ func (notes ReleaseNotes) Print(milestone string) error {
 	var none []ReleaseNote
 	for _, n := range notes {
 		switch n.Typology {
+		case "BREAKING CHANGE":
+			breaking = append(breaking, n)
 		case "fix":
 			fixes = append(fixes, n)
-			break
```
Comment (FedeDP) [inline pkg/releasenotes/print.go:85]: No need to break here.

File: pkg/releasenotes/print.go [hunk 2]
```diff
@@ -80,24 +80,18 @@ func (notes ReleaseNotes) Print(milestone string) error {
...
 		case "rule":
 			rules = append(rules, n)
-			break
-		case "BREAKING CHANGE":
-			breaking = append(breaking, n)
-			break
-		case "new":
+		case "new", "feat":
```
Comment (FedeDP) [inline pkg/releasenotes/print.go:89]: Support "feat" too!

File: pkg/releasenotes/print.go [hunk 3]
```diff
@@ -138,8 +132,8 @@ func (s *Statistics) Print() error {
 	table.SetBorders(tablewriter.Border{Left: true, Top: false, Right: true, Bottom: false})
 	table.SetCenterSeparator("|")
 
-	table.Append([]string{"Not user-facing", strconv.FormatInt(s.totalNone, 10)})
-	table.Append([]string{"Release note", strconv.FormatInt(s.total-s.totalNone, 10)})
+	table.Append([]string{"Not user-facing", strconv.FormatInt(s.nonFacing, 10)})
```
Comment (FedeDP) [inline pkg/releasenotes/print.go:135]: Just a rename.

File: pkg/releasenotes/releasenotes.go [hunk 4]
```diff
@@ -13,7 +13,7 @@ import (
 
 var (
 	releaseNoteRegexp = regexp.MustCompile("(?s)```release-note(.+?)```")
-	typologyRegexp    = regexp.MustCompile(`(?m)(.+?)(\((.+)\))?: ?(.*)`)
+	typologyRegexp    = regexp.MustCompile(`(?m)(.+?)(\((.+)\))?(!)?: ?(.*)`)
```
Comment (FedeDP) [inline pkg/releasenotes/releasenotes.go:16]: Support `!` in the regex.

File: pkg/releasenotes/releasenotes.go [hunk 5]
```diff
@@ -137,6 +138,9 @@ func (c *Client) Get(org, repo, branch, milestone string) (ReleaseNotes, *Statis
 				Author:      fmt.Sprintf("@%s", p.GetUser().GetLogin()),
 				AuthorURL:   p.GetUser().GetHTMLURL(),
 			}
+			if matches[4] == "!" {
+				rn.Typology = "BREAKING CHANGE"
```
Comment (FedeDP) [inline pkg/releasenotes/releasenotes.go:142]: If `!` exists, this is a BREAKING CHANGE.

**Comment by FedeDP** [comment]
Btw first  Falco release with the rn2md automation is out: https://github.com/falcosecurity/falco/releases/tag/0.36.0 :smile:

**Comment by FedeDP** [comment]
Also, i was thinking about moving "breaking changes" to their own markdown section; wdyt?

File: pkg/releasenotes/releasenotes.go (hunk 5 above)
Comment (leodido) [reply pkg/releasenotes/releasenotes.go:142]: Correct!

**Comment by leodido** [comment]
> Also, i was thinking about moving "breaking changes" to their own markdown section; wdyt?

Ok, but what about making it configurable?

**Review by leodido — APPROVED** [review]

**Review by leodido — APPROVED** [review]

File: pkg/releasenotes/releasenotes.go (hunk 4 above)
Comment (leodido) [reply pkg/releasenotes/releasenotes.go:16]: You've just reminded me that ideally it would be better to integrate go-conventialcommits. See issue #3 :)

**Comment by FedeDP** [comment]
> Ok, but what about making it configurable?

Love it! What about making configurable the entire template? I mean, we ship the default one, but users can override it with their own supplied template (a simple txt file). 
And the action would allow it to be passed as input parameter obviously.

**Comment by FedeDP** [comment]
Allowing people to override the default template would require the templated strings to become an API though; i am not sure we want it; for example, `MajorNotes` or `BreakingNotes` would become public interface.

**Pull request merged**